frontend/src/App.tsx

```diff
@@ -555,9 +555,23 @@ export class App extends PureComponent<Props, State> {
         )
       }
 
+      // NOTE: This code (and the !themeInput case below) needs to be fixed to
+      // handle the case where the user has their theme preference set to
+      // Auto/Dark/Light and we don't want to ignore that.
```

Comment:
I'm mixed. What if the custom theme is new and user already had a preference? Then they would never end up seeing the custom theme added by the developer. Ideally we only update if previously there was no custom theme or had an old theme, then we fall back onto their preference.

Now that I'm thinking about it, I kinda of just want to remove light and dark and just have auto. Could just be me but I'm thinking why would someone using dark mode want to use light mode for an app? Though I guess if the developer using custom components that aren't theme friendly, I could see switching to light mode as an fix but would users really know to 'debug' that and do so?

Reply:
The user setting a preference early so never seeing a theme added later (or changes to the an existing theme) was another edge case that crossed my mind. I'll probably just write up a list of interactions that can happen so product can think things over since the ideal solution is probably a decent amount of work to implement.

I think Light/Dark are probably still necessary for people running OSes that don't have a system-wide toggle (older versions of windows/MacOS, lots of linux distros), but yeah I do think if it weren't for that then cases where a dark-mode person would specifically want certain streamlit apps to be light mode would be rare.